### ISSUE TITLE
terser plugin doesn't pass check by typescript (TS1479)

### ISSUE THREAD

**Issue by eXory2024** [rollup/plugins#1795]
- Rollup Plugin Name:  terser
- Rollup Plugin Version: 0.4.4
- Rollup Version: 4.24.0
- Operating System (or Browser): MacOS
- Node Version: 20.11.1
- Link to reproduction _(⚠️ read below)_: https://stackblitz.com/edit/rollup-repro-5ppdqp

### Expected Behavior

No error should be emitted by tsc.

### Actual Behavior

The following error is emitted by tsc:

```
error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("terser")' call instead.

2 import type { MinifyOptions } from 'terser';
                                     ~~~~~~~~


Found 1 error in node_modules/@rollup/plugin-terser/types/index.d.ts:2
```

### Additional Information

The rollup config is intentionally not populated as it is not relevant to the bug / behaviour.

**Comment by notaphplover** [comment]
Yeah, https://arethetypeswrong.github.io/?p=%40rollup%2Fplugin-terser%400.4.4. Are you open to accept PR?

**Comment by eXory2024** [comment]
I would if I knew what had to be done exactly.

It looks to me the issue arises from having only one d.ts file from both the cjs and es export.

The website you posted links to a document that suggests defining a separate d.ts file for the cjs and es export like so:


```json
{
  "name": "pkg",
  "exports": {
    ".": {
      "import": {
        "types": "./index.d.mts",
        "default": "./index.mjs"
      },
      "require": {
        "types": "./index.d.ts",
        "default": "./index.js"
      }
    }
  }
}
```

However this would mean duplicating the index.d.ts file since (from what I can tell) it is handwritten and not automatically generated.

I really don't know much about how TypeScript handles CommonJS to be able to formulate a solution in my own time.
